client/src/auth/emailPassword/supabase.email-password-provider.ts

```diff
@@ -0,0 +1,21 @@
+import { supabase } from "../../supabase";
+import { EmailPassword } from "./email-password.interface";
+
+export class SupabaseEmailPasswordProvider implements EmailPassword {
+  async signUp(email: string, password: string): Promise<boolean> {
+    const { error } = await supabase.auth.signUp({
+      email,
+      password,
+    });
+
+    return !!error;
```

Comment:
use lodash functions (isUndefined), dont abuse !!, makes it more readable and avoids weird JS behavior (same for all places with !! or ! on non boolean variables)

Reply:
Can you be more specific about the problem that can be caused? This object is either null or object, so it should be fine in my opinion. I can use the function Boolean if you think it could be more readable

Reply:
maybe here it wont cause too much problems but its just a bad habit, and for readability sake I would still use isEmpty from lodash which is basicly doing !!, just makes it readable

Reply:
fixed